Menus/ScenarioActivityConfigGUI.cpp

```diff
@@ -0,0 +1,445 @@
+			if (m_SelectedActivity->TeamActive(team)) {
+				teamIcon = m_SelectedActivity->GetTeamIcon(team);
+				if (!teamIcon) { teamIcon = dynamic_cast<const Icon *>(g_PresetMan.GetEntityPreset("Icon", "Team " + std::to_string(team + 1) + " Default")); }
+
+				m_TeamNameLabels.at(team)->SetText(m_SelectedActivity->GetTeamName(team));
+				m_TeamTechComboBoxes.at(team)->SetVisible(true);
+				m_TeamAISkillSliders.at(team)->SetVisible(true);
+				m_TeamAISkillLabels.at(team)->SetVisible(true);
+			} else {
+				teamIcon = dynamic_cast<const Icon *>(g_PresetMan.GetEntityPreset("Icon", "Locked Team"));
+				m_TeamNameLabels.at(team)->SetText("Unavailable");
+				m_TeamTechComboBoxes.at(team)->SetVisible(false);
+				m_TeamAISkillSliders.at(team)->SetVisible(false);
+				m_TeamAISkillLabels.at(team)->SetVisible(false);
+			}
+			if (teamIcon) { m_TeamIconBoxes.at(team)->SetDrawImage(new AllegroBitmap(teamIcon->GetBitmaps32()[0])); }
+		}
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void ScenarioActivityConfigGUI::StartGame() {
+		GameActivity *gameActivity = dynamic_cast<GameActivity *>(m_SelectedActivity->Clone());
+
+		gameActivity->SetDifficulty(m_ActivityDifficultySlider->GetValue());
+		gameActivity->SetStartingGold((m_StartingGoldSlider->GetValue() == m_StartingGoldSlider->GetMaximum()) ? 9999999 : m_StartingGoldSlider->GetValue() - (m_StartingGoldSlider->GetValue() % 500));
+
+		gameActivity->SetRequireClearPathToOrbit(m_RequireClearPathToOrbitCheckbox->GetCheck());
+		gameActivity->SetFogOfWarEnabled(m_FogOfWarCheckbox->GetCheck());
+		g_SceneMan.SetSceneToLoad(m_SelectedScene, true, m_DeployUnitsCheckbox->GetCheck());
+
+		gameActivity->ClearPlayers(false);
+		for (int player = Players::PlayerOne; player < Players::MaxPlayerCount; ++player) {
+			for (int team = Activity::Teams::TeamOne; team < Activity::Teams::MaxTeamCount; ++team) {
+				if (m_PlayerBoxes.at(player).at(team)->GetDrawType() == GUICollectionBox::Image) {
+					gameActivity->AddPlayer(player, true, team, 0);
+					break;
+				}
+			}
+		}
+		for (int team = Activity::Teams::TeamOne; team < Activity::Teams::MaxTeamCount; ++team) {
+			if (m_PlayerBoxes.at(PlayerColumns::PlayerCPU).at(team)->GetDrawType() == GUICollectionBox::Image) {
+				gameActivity->SetCPUTeam(team);
+				break;
+			}
+		}
+		for (int team = Activity::Teams::TeamOne; team < Activity::Teams::MaxTeamCount; ++team) {
+			if (const GUIListPanel::Item *techItem = m_TeamTechComboBoxes.at(team)->GetSelectedItem()) {
+				if (techItem->m_ExtraIndex == -2) {
+					gameActivity->SetTeamTech(team, "-All-");
+				} else if (techItem->m_ExtraIndex == -1) {
+					m_TeamTechComboBoxes.at(team)->SetSelectedIndex(RandomNum<int>(2, m_TeamTechComboBoxes.at(team)->GetListPanel()->GetItemList()->size() - 1));
+					gameActivity->SetTeamTech(team, g_PresetMan.GetDataModuleName(m_TeamTechComboBoxes.at(team)->GetSelectedItem()->m_ExtraIndex));
```

Comment:
Wuh.
`techItem` is a pointer to a list entry in the combobox.

Reply:
Oh whoops, my bad, I missed the fact that the SetSelectedIndex line was for `m_TeamTechComboBoxes.at(team)` rather than `m_TeamTechComboBoxes.at(team)->GetSelectedItem()`. Sorry for the confusion lol

Btw, here's a corrected version of the rewrite that I put together to understand what's going on/what I was thinking. Feel free to use it or keep what you have, I think either version is fine:
```cpp
if (const GUIListPanel::Item *techItem = m_TeamTechComboBoxes.at(team)->GetSelectedItem()) {
	if (techItem->m_ExtraIndex == -1) {
		m_TeamTechComboBoxes.at(team)->SetSelectedIndex(RandomNum<int>(2, m_TeamTechComboBoxes.at(team)->GetListPanel()->GetItemList()->size() - 1);
		techItem = m_TeamTechComboBoxes.at(team)->GetSelectedItem();
	}
	if (techItem->m_ExtraIndex == -2) {
		gameActivity->SetTeamTech(team, "-All-");
	} else {
		gameActivity->SetTeamTech(team, g_PresetMan.GetDataModuleName(techItem->m_ExtraIndex));
	}
}
```